lab-integration/src/main.ts

```diff
@@ -1,47 +1,90 @@
+import { WatchEventType, promises as fs, watch } from "fs";
+import * as lockfile from "proper-lockfile";
 import path from "path";
+import { queue } from "async";
+import puppeteer from "puppeteer";
+
+import { DateTime } from "luxon";
 import { TestsResult } from "./types";
 
 const filePath = path.join(process.cwd(), "../test-results.json");
 
+const machines = new Map<string, string>([
+  ["5534", "1 - Prensa PRE-01"],
+  ["5533", "2 - Prensa PRE-02"],
+  ["5532", "3 - Prensa PRE-03"],
+]);
+
 (async () => {
   console.info("SCRIPT IS RUNNING...");
 
   let lastTestResultDate = DateTime.now();
 
-  watch(filePath, async (eventType) => {
-    const stringifiedDataset = await fs.readFile(filePath, "utf8");
-    const parsedDataset = JSON.parse(stringifiedDataset) as TestsResult[];
-    const dataset = parsedDataset
-      .map((testResult) => {
-        const Data = DateTime.fromFormat(
-          testResult.Data,
-          "dd/LL/yyyy, HH:mm:ss",
-          { locale: "pt-BR" },
-        );
-
-        return {
-          ...testResult,
-          Data,
-        };
-      })
-      .filter((testResult) => testResult.Data.isValid);
-
-    if (eventType === "change") {
-      const newTestsResults = dataset.filter(
-        (testResult) =>
-          testResult.Data.diff(lastTestResultDate).milliseconds > 0,
-      );
-
-      if (newTestsResults.length > 0) {
-        lastTestResultDate = dataset.sort(
-          (a, b) => b.Data.diff(a.Data).milliseconds,
-        )[0].Data as unknown as DateTime<true>;
+  const browser = await puppeteer.launch({ headless: false, slowMo: 100 });
+  const page = await browser.newPage();
+  await page.setViewport({ width: 720, height: 720 });
+  await page.goto("http://localhost:3000/submission");
+
+  const queueTask = queue(
+    async (task: { eventType: WatchEventType }, callback) => {
+      try {
+        if (task.eventType === "change") {
+          await lockfile.lock(filePath);
+          const stringifiedDataset = await fs.readFile(filePath, "utf8");
+          const parsedDataset = JSON.parse(stringifiedDataset) as TestsResult[];
+          const dataset = parsedDataset
+            .map((testResult) => {
+              const Data = DateTime.fromFormat(
+                testResult.Data,
+                "dd/LL/yyyy, HH:mm:ss",
+                { locale: "pt-BR" },
+              );
+
+              return {
+                ...testResult,
+                Data,
+              };
+            })
+            .filter((testResult) => testResult.Data.isValid);
+
+          const newTestsResults = dataset.filter(
+            (testResult) =>
+              testResult.Data.diff(lastTestResultDate).milliseconds > 0,
+          );
+          if (newTestsResults.length > 0) {
+            lastTestResultDate = dataset.sort(
+              (a, b) => b.Data.diff(a.Data).milliseconds,
+            )[0].Data as DateTime<true>;
+
+            for (const testResult of newTestsResults) {
```

Comment:
Let's use `forEach` because it's more concise.

Reply:
We couldn't 'cause the promises. Whether we do the .forEach method won't wait the promise finish and will start the next loop.

As well you want to use **Promisse.All** it doesn't gonna work too, because it will start all promises and do not will wait each puppeeter action finishes sequentially to start another breaking the next one that needs the action before be completed.